go-controller/pkg/factory/handler.go

```diff
@@ -84,6 +91,55 @@ func (h *Handler) kill() bool {
 	return atomic.CompareAndSwapUint32(&h.tombstone, handlerAlive, handlerDead)
 }
 
+// handleEvent checks if this handler is enabled, if not it buffers the event for processing later.
+func (h *Handler) handleEvent(processEvent func()) {
+	if atomic.LoadUint32(&h.tombstone) != handlerAlive {
+		// handler is dead
+		return
+	}
+	// Fast path: handler is enabled, no buffering
+	if h.enabled.Load() {
+		processEvent()
+		return
+	}
+	// Slow path: still disabled, might need to buffer
+	h.mu.Lock()
+	if !h.enabled.Load() {
+		h.buffer = append(h.buffer, processEvent)
```

Comment:
I think this breaks `forEachQueuedHandler` protocol: it is supposed to sequentially run handlers based on their priority, but now if you have disabled handlers they will be skipped, and then they will handle the same event later, so the priorities of handlers don't work anymore. Especially considering the handler could forever (or for a long time) stay disabled?

Reply:
Another part of this is adding handlers for the same object type but with different priorities: e.g. we add podHandler first, then we start netpol handlers watching pod labels. Before this change you know that when the `AddPodHandler` returns, the handler is added (and enabled) and when you call `AddNetpolPodHandler` after that, you know that every event will be handled by the PodHandler first, and only then passed to the netpol handler. This contract also won't work anymore, and I can imagine it being a problem.